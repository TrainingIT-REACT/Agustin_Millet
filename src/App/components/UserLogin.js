import React, { useState, useContext, Fragment } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import { Button, Typography } from "@material-ui/core";
import UserContext from "../contexts/user";
import { Redirect } from "react-router";

const styles = theme => ({
  container: {
    display: "flex",
    paddingLeft: "36%",
    flexWrap: "wrap",
    backgroundColor: theme.palette.background.paper
  },
  textField: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1)
  },
  button: {
    marginLeft: theme.spacing(1),
    marginRight: theme.spacing(1),
    width: 223,
    height: 50
  },
  buttonContainer: {
    width: "62%",
    marginTop: 9
  },
  message: {
    marginTop: 18
  },
  deniedMessage: {
    marginTop: 20,
    color: "red"
  }
});

const UserLogin = ({ classes, location }) => {
  const [login, setLogin] = useState({
    login: "",
    password: "",
    errorLogin: false,
    errorPwd: false,
    denied: false
  });

  //const [signedIn, setSignedIn] = useState(false);

  const updateLoginState = (field, value) => {
    login[field] = value;
    setLogin({ ...login });
  };

  let userContext = useContext(UserContext);

  const handleChange = field => event => {
    updateLoginState(field, event.target.value);
  };

  const handleLogin = e => {
    setLogin({
      ...login,
      errorLogin: login.login === "",
      errorPwd: login.password === "",
      denied: !(login.login === "test" && login.password === "pwd")
    });
    userContext.signedIn = login.login === "test" && login.password === "pwd";
  };

  return (
    <form className={classes.container} noValidate autoComplete="off">
      <div>
        {userContext.signedIn && (
          <Typography
            className={classes.message}
            variant="subtitle1"
            color="textSecondary"
          >
            Usuario identificado correctamente
          </Typography>
        )}
        {!userContext.signedIn && (
          <Fragment>
            {location.state && location.state.redirected && (
              <Typography
                className={classes.message}
                variant="subtitle1"
                color="textSecondary"
              >
                Estás intentando acceder a un área privada. Por favor, haz login
                primero.
              </Typography>
            )}
            {login.denied && (
              <Typography className={classes.deniedMessage} variant="subtitle1">
                Usuario y/o contraseña incorrectos
              </Typography>
            )}
            <TextField
              required
              error={login.errorLogin}
              id="outlined-login"
              label="Login"
              className={classes.textField}
              value={login.login}
              onChange={handleChange("login")}
              margin="normal"
              variant="outlined"
            />
            <TextField
              required
              error={login.errorPwd}
              id="outlined-password-input"
              label="Password"
              className={classes.textField}
              value={login.password}
              onChange={handleChange("password")}
              type="password"
              autoComplete="current-password"
              margin="normal"
              variant="outlined"
            />
          </Fragment>
        )}{" "}
      </div>

      {!userContext.signedIn && (
        <div className={classes.buttonContainer}>
          <Button
            onClick={handleLogin}
            variant="contained"
            color="primary"
            className={classes.button}
          >
            Acceder
          </Button>
        </div>
      )}

      {userContext.signedIn === true &&
        location.state &&
        location.state.protectedPath && (
          <Redirect to={location.state.protectedPath} />
        )}
    </form>
  );
};

UserLogin.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(UserLogin);
