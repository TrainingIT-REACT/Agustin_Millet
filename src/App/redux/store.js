import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import promise from "redux-promise-middleware";
import login from "./reducers/loginReducer";
import navigation from "./reducers/navigationReducer";
import profile from "./reducers/profileReducer";
import albums from "./reducers/albumsReducer";
import songs from "./reducers/songsReducer";

// For integrating REDUX DEVTOOLS
// Set up our composeEnhancers function, baed on the existence of the
// DevTools extension when creating the store
let storeTemp;
if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) {
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
  storeTemp = createStore(
    combineReducers({ login, navigation, profile, albums, songs }),
    undefined, // or initial state provided by startup
    composeEnhancers(applyMiddleware(promise))
  );
} else {
  storeTemp = createStore(
    combineReducers({ login, navigation, profile, albums, songs }),
    applyMiddleware(promise)
  );
}

const store = { ...storeTemp };

export default store;
