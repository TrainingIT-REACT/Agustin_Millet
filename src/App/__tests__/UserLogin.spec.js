import React, { Fragment } from "react";
import { render, shallow, mount } from "enzyme";
import UserLogin from "../components/UserLogin";
import { Provider } from "react-redux";
import store from "../redux/store";

/**
 * Testing de 2 componentes de presentaciÃ³n
 */

/**
 * UserLogin
 */
describe('UserLogin', () => {

  describe("Render", () => {
      const wrapper = render(<Provider store={store}>
        <UserLogin location={ { state: { redirected: false } }}/>
        </Provider>
      );


      it('should match the following texts', () => {
      // Comprobamos el texto
      expect(wrapper.find('label[for="outlined-login"]').text()).toBe("Loginâ*");
      expect(wrapper.find('label[for="outlined-password-input"]').text()).toBe("Passwordâ*");

  
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot(); 
  });        
  });

  describe("Mount", () => {
    const wrapper = mount(<Provider store={store}>
      <UserLogin location={ { state: { redirected: false } }}/>
      </Provider>
    );

    it("should change class on empty fields", () => {
      const button = wrapper.find('button')

      button.simulate('click')
      expect(wrapper.find('div.Mui-error').length).toBe(2);

      });
});  






  });

 


