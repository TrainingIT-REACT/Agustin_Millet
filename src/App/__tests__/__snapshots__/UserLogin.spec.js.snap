// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLogin Render should match the snapshot 1`] = `
<form
  autocomplete="off"
  class="UserLogin-container-1"
  novalidate=""
>
  <div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root UserLogin-textField-2"
    >
      <label
        class="MuiFormLabel-root Mui-required Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
        for="outlined-login"
      >
        Login
        <span
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-60 MuiOutlinedInput-notchedOutline"
          style="padding-left:8px"
        >
          <legend
            class="PrivateNotchedOutline-legend-61"
            style="width:0.01px"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="outlined-login"
          required=""
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root UserLogin-textField-2"
    >
      <label
        class="MuiFormLabel-root Mui-required Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink="false"
        for="outlined-password-input"
      >
        Password
        <span
          class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           *
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-60 MuiOutlinedInput-notchedOutline"
          style="padding-left:8px"
        >
          <legend
            class="PrivateNotchedOutline-legend-61"
            style="width:0.01px"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <input
          aria-invalid="false"
          autocomplete="current-password"
          class="MuiInputBase-input MuiOutlinedInput-input"
          id="outlined-password-input"
          required=""
          type="password"
          value=""
        />
      </div>
    </div>
     
  </div>
  <div
    class="UserLogin-buttonContainer-4"
  >
    <button
      class="MuiButtonBase-root MuiButton-root UserLogin-button-3 MuiButton-containedPrimary MuiButton-contained"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Acceder
      </span>
    </button>
  </div>
</form>
`;
